tests/bearlib/abstractions/LinterTest.py

```diff
@@ -328,6 +347,50 @@ def test_process_output_corrected(self):
                                           original))
         self.assertEqual(len(results), 2)
 
+    def test_process_output_unidiff(self):
+        uut = (linter(sys.executable, output_format='unified-diff')
+               (self.EmptyTestLinter)
+               (self.section, None))
+
+        original = ['void main()  {', 'return 09;', '}']
+
+        diff = ['--- a/some-file.c',
+                '+++ b/some-file.c',
+                '@@ -1,3 +1,4 @@',
+                '-void main() {',
+                '-       return 09;',
+                '+void main()'
+                '+{',
+                '+       return 9;',
+                ' }']
+        diff_string = '\n'.join(diff)
+
+        results = list(uut.process_output(diff_string,
+                                          'some-file.c',
+                                          original))
+
+        diffs = list(Diff.from_unified_diff(diff_string, original).split_diff())
+
+        expected = [Result.from_values(uut,
+                                       'Inconsistency found.',
+                                       'some-file.c',
+                                       1, None, 2, None,
+                                       RESULT_SEVERITY.NORMAL,
+                                       diffs={'some-file.c': diffs[0]})]
+
+        self.assertEqual(results, expected)
+
+        uut = (linter(sys.executable,
+                      output_format='unified-diff',
+                      diff_distance=-1)
+               (self.EmptyTestLinter)
+               (self.section, None))
+
+        results = list(uut.process_output(diff_string,
+                                          'some-file.c',
+                                          original))
+        self.assertEqual(len(results), 2)
```

Comment:
Yeah, there should be one. Also, I've added tests for unified diffs with some unchanged part of files missing in `DiffTest.py` as well.